interview-questions.md

```diff
@@ -7,46 +7,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
+  Your answer: The name of the foreign key would be "animal_id" and it would belong to the Sightings model. The foreign key is always in the belongs_to side.
 
-  Researched answer:
+  Researched answer: It should be "animal_id:integer"
 
 
 
 2. Which routes must always be passed params and why?
 
-  Your answer:
+  Your answer: DELELTE, EDIT, UPDATE and SHOW routes must always have params pased because you hvae to specify which of the many Objects of a model, or which parameter of many parameters, you are planning to work with.
 
-  Researched answer:
+  Researched answer: Same
 
 
 
 3. Based on your knowledge of Rails conventions, write a route for a webpage called "game" that takes in a parameter called "guess" with a controller called "main".
 
-  Your answer:
+  Your answer: get 'game/:guess' => 'main#game' 
 
 
 
 4. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: 
+  1) rails g new model Example attribute1:integer attribute2:string
+  This creates a new model to store data in. It will have two columns, one that takes an integer value and one that takes in a string value. 
+
+  2) rails g new controller Example 
+  This creates  controller file for the Example model, which handles how different actions are handled and displayed.
+
+  3) rails g new migration Example 'change_integer_to_string_of_attribute1'
```

Comment:
Totally! I would mention the proper naming convention here of starting with the verb, and ending with the model it is effecting. 

(I know you got to this in your researched answer though!)